praw/helpers.py

```diff
@@ -111,6 +111,104 @@ def valid_redditors(redditors, sub):
+        unix timestamps(just like ``created_utc`` attributes)
+    :param verbosity: A number that controls the amount of output produced to
+        stderr. <= 0: no output; >= 1: output the total number of submissions
+        processed; >= 2: output debugging information regarding
+        the search queries. (Default: 1)
+
+    """
+    def debug(msg, level):
+        if verbosity >= level:
+            sys.stderr.write(msg + '\n')
+
+    reddit_timestamp_offset = 28800
+    if highest_timestamp is None:
+        # convert normal unix timestamps into broken reddit timestamps
+        highest_timestamp = int(time.time()) + reddit_timestamp_offset
+    else:
+        highest_timestamp += reddit_timestamp_offset
+
+    if lowest_timestamp is not None:
+        lowest_timestamp += reddit_timestamp_offset
+    elif not isinstance(subreddit, six.string_types):
+        lowest_timestamp = subreddit.created
+    elif subreddit not in ("all", "contrib", "mod", "friend"):
+        lowest_timestamp = reddit_session.get_subreddit(subreddit).created
+    else:
+        lowest_timestamp = 0
+
+    # Those parameters work ok, but there may be a better set of parameters
+    window_size = 60 * 60
+    search_limit = 100
+    min_search_results_in_window = 30
+    backoff = BACKOFF_START
+
+    processed_submissions = 0
+    while highest_timestamp >= lowest_timestamp:
+        try:
+            search_query = 'timestamp:{}..{}'.format(
+                max(highest_timestamp - window_size, lowest_timestamp),
+                highest_timestamp)
+            debug(search_query, 3)
+            search_results = list(reddit_session.search(search_query,
+                                                        subreddit=subreddit,
+                                                        limit=search_limit,
+                                                        syntax='cloudsearch',
+                                                        sort='new'))
+
+            debug("Received {0} search results for query {1}"
+                  .format(len(search_results), search_query), 2)
+
+            backoff = BACKOFF_START
+        except HTTPException as exc:
+            debug("{0}. Sleeping for {1} seconds".format(exc, backoff), 2)
+            time.sleep(backoff)
+            backoff *= 2
+            continue
+
+        if len(search_results) >= search_limit:
+            window_size /= 2
+            debug("Reducing window size to {0} seconds".format(window_size), 2)
+            continue
+
+        for submission in search_results:
+            yield submission
+
+        highest_timestamp -= (window_size + 1)
+        processed_submissions += len(search_results)
+        debug('Total processed submissions: {}'
+              .format(processed_submissions), 1)
+
+        if len(search_results) < min_search_results_in_window:
+            debug("Increasing window size to {0} seconds"
+                  .format(window_size), 2)
+            window_size *= 2
```

Comment:
Should this statement occur before line 207 for the debug statement to be accurate?